internal/transform/operation.go

```diff
@@ -1690,26 +1656,38 @@ func (operation *transactionOperationWrapper) Details() (map[string]interface{},
 			details["contract_id"] = contractIdFromTxEnvelope(transactionEnvelope)
 			details["contract_code_hash"] = contractCodeHashFromTxEnvelope(transactionEnvelope)
 
-			switch args.ContractIdPreimage.Type {
-			case xdr.ContractIdPreimageTypeContractIdPreimageFromAddress:
-				fromAddress := args.ContractIdPreimage.MustFromAddress()
-				address, err := fromAddress.Address.String()
-				if err != nil {
-					panic(fmt.Errorf("error obtaining address for: %s", args.ContractIdPreimage.Type))
+			preimageTypeMap := switchContractIdPreimageType(args.ContractIdPreimage)
+			for key, val := range preimageTypeMap {
+				if _, ok := preimageTypeMap[key]; ok {
+					details[key] = val
 				}
-				details["from"] = "address"
-				details["address"] = address
-			case xdr.ContractIdPreimageTypeContractIdPreimageFromAsset:
-				details["from"] = "asset"
-				details["asset"] = args.ContractIdPreimage.MustFromAsset().StringCanonical()
-			default:
-				panic(fmt.Errorf("unknown contract id type: %s", args.ContractIdPreimage.Type))
 			}
 		case xdr.HostFunctionTypeHostFunctionTypeUploadContractWasm:
 			details["type"] = "upload_wasm"
 			transactionEnvelope := getTransactionV1Envelope(operation.transaction.Envelope)
 			details["ledger_key_hash"] = ledgerKeyHashFromTxEnvelope(transactionEnvelope)
 			details["contract_code_hash"] = contractCodeHashFromTxEnvelope(transactionEnvelope)
+		case xdr.HostFunctionTypeHostFunctionTypeCreateContractV2:
+			args := op.HostFunction.MustCreateContractV2()
+			details["type"] = "create_contract_v2"
+
+			transactionEnvelope := getTransactionV1Envelope(operation.transaction.Envelope)
+			details["ledger_key_hash"] = ledgerKeyHashFromTxEnvelope(transactionEnvelope)
+			details["contract_id"] = contractIdFromTxEnvelope(transactionEnvelope)
+			details["contract_code_hash"] = contractCodeHashFromTxEnvelope(transactionEnvelope)
+
+			// ConstructorArgs is a list of ScVals
+			// This will initially be handled the same as InvokeContractParams until a different
+			// model is found necessary.
```

Comment:
when you say "until a different model is found necessary" do you mean if we need to update the schema/data model? Or is there a possibility core will change how they present this data?